pkg/cnb/project_descriptor.go

```diff
@@ -0,0 +1,89 @@
+package cnb
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+	ignore "github.com/sabhiram/go-gitignore"
+)
+
+const fileDescriptorName = "project.toml"
+
+func ProcessProjectDescriptor(appDir, platformDir string) error {
+	var d descriptor
+	file := filepath.Join(appDir, fileDescriptorName)
+	if _, err := os.Stat(file); os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("unable to determine if project descriptor file exists: %w", err)
+	}
+	_, err := toml.DecodeFile(file, &d)
+	if err != nil {
+		return err
+	}
+	if err := processFiles(appDir, d); err != nil {
+		return err
+	}
+	return serializeEnvVars(d.Build.Env, platformDir)
+}
+
+func processFiles(appDir string, d descriptor) error {
+	fileFilter, err := getFileFilter(d)
+	if err != nil {
+		return err
+	}
+	if fileFilter == nil {
+		return nil
+	}
+	return filepath.Walk(appDir, func(path string, f os.FileInfo, err error) error {
+		if err != nil {
+			return nil
```

Comment:
Should this return the `err`?

Reply:
It seems like if we run into an error walking the directory we might not be able to accurately apply the includes/excludes and we should error out?

Reply:
Makes sense. Will update :)

Reply:
Should be fixed now :)